src/front-end/components/game/skills/skill-button.tsx

```diff
@@ -0,0 +1,28 @@
+import React, { type FC, useMemo } from 'react';
+import { ProgressBar } from '@/front-end/components/ui/progress-bar.tsx';
+import { Column } from '@/front-end/components/layout/column.tsx';
+import { Typography } from '@/front-end/components/ui/typography.tsx';
+import type { Skill } from '@/shared/definition/schema/types/types-skills.ts';
+import { xpAccum, xpToReachNext } from '@/shared/util/util-skills.ts';
+
+interface Props {
+  skill: Skill;
+}
+
+export const SkillButton: FC<Props> = React.memo(function SkillButton(props) {
+  const { skill } = props;
+
+  const skillXp = useMemo(
+    () => (skill.level === 0 ? skill.xp : skill.xp - xpAccum[skill.level - 1]),
+    [skill.level, skill.xp],
+  );
+  const targetXp = useMemo(() => xpToReachNext(skill.level), [skill.level]);
```

Comment:
I would have used total xp amounts instead of doing it like this. Maybe we could turn that into a setting later if you dont agree?